core/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -119,7 +121,9 @@ private static ClientTemplate buildTemplate(Settings providedSettings, Settings
             }
             SettingsModule settingsModule = new SettingsModule(settings, additionalSettings, additionalSettingsFilter);
 
-            NetworkModule networkModule = new NetworkModule(networkService, settings, true);
+            NetworkModule networkModule = new NetworkModule(settings, true);
+            networkModule.processPlugins(pluginsService.filterPlugins(NetworkPlugin.class));
```

Comment:
Could you make the plugins a ctor argument like the other modules? Making the modules immutable makes them much easier to understand. We could do it in a followup/not at all if that is too strange for this module.